Correct format for product_recall_alert_date

beccapearce opened the pull request:
This field wasn't having any effect because it was specified to have a
format of oim_project - presumably this was a copy and paste error.

This was causing the product_recall_alert_date to not show up in the finder.

kevindew commented on spec/features/creating_a_product_safety_alert_report_recall.rb:
```diff
@@ -0,0 +1,103 @@
+require "spec_helper"
+
+RSpec.feature "Creating a DRCF digital markets research", type: :feature do
```
Ooh copy+paste issue to fix ☝️

beccapearce committed:
Correct format for product_recall_alert_date

This field wasn't having any effect because it was specified to have a
format of oim_project - presumably this was a copy and paste error.

Add a test to ensure that the error is caught if it should happen again.

kevindew reviewed (approved):
Super
